File: client/src/types/JsonModels/Subscriptions/SubscriptionPlanModel.ts
import type { JsonModels } from "../../backend";

export interface SubscriptionPlanModel
{
	id: number;
	code: string;
	name: string;
	description?: string;
	sortOrder: number;
	prices: JsonModels.Subscriptions.SubscriptionPlanPriceModel[];
	features: JsonModels.Subscriptions.PlanFeatureModel[];
}

File: server/FitMate.Core/JsonModels/Subscriptions/SubscriptionPlanModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.Subscriptions;

public class SubscriptionPlanModel
{
    public long Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int SortOrder { get; set; }
    public List<SubscriptionPlanPriceModel> Prices { get; set; } = [];
    public List<PlanFeatureModel> Features { get; set; } = [];
}
